src/service/action-impl.js

```diff
@@ -566,20 +583,52 @@ export class ActionService {
    * @param {!Array<{tagOrTarget: string, method: string}>} whitelist
    */
   setWhitelist(whitelist) {
-    this.whitelist_ = whitelist;
+    this.whitelist_ = whitelist.filter((value) => {
+      const hasValidProperties = value.tagOrTarget && value.method;
+      if (!hasValidProperties) {
+        user().warn(
+          TAG_,
+          'Unexpected entry missing properties `tagOrTarget` and `method` was not whitelisted: %s',
+          value
+        );
+      }
+      return hasValidProperties;
+    });
   }
 
   /**
    * Adds an action to the whitelist.
    * @param {string} tagOrTarget The tag or target to whitelist, e.g.
    *     'AMP-LIST', '*'.
-   * @param {string} method The method to whitelist, e.g. 'show', 'hide'.
+   * @param {string|Array<string>} methods The method(s) to whitelist, e.g. 'show', 'hide'.
+   * @param {Array<string>=} opt_allowedFormats
    */
-  addToWhitelist(tagOrTarget, method) {
+  addToWhitelist(tagOrTarget, methods, opt_allowedFormats) {
```

Comment:
Nit: `opt_forFormat` so the function signature would read "adds to whitelist... tag.method... for format f". 

"Allowed format" makes me think "which formats are not allowed?".